x-pack/platform/plugins/private/intercepts/server/services/intercept_user_interaction.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import assert from 'assert';
+import {
+  type CoreSetup,
+  type CoreStart,
+  type Logger,
+  RequestHandlerContext,
+  SavedObjectsErrorHelpers,
+} from '@kbn/core/server';
+import { schema } from '@kbn/config-schema';
+import type { ISavedObjectsRepository, SavedObject } from '@kbn/core-saved-objects-api-server';
+import {
+  interceptInteractionUserRecordSavedObject,
+  type InterceptInteractionUserRecordAttributes,
+} from '../saved_objects';
+import { TRIGGER_USER_INTERACTION_METADATA_API_ROUTE } from '../../common/constants';
+
+export class InterceptUserInteractionService {
+  private savedObjectsClient?: ISavedObjectsRepository;
+  private savedObjectRef = interceptInteractionUserRecordSavedObject;
+
+  setup(core: CoreSetup, logger: Logger) {
+    core.savedObjects.registerType(this.savedObjectRef);
+
+    const router = core.http.createRouter<RequestHandlerContext>();
+
+    router.get(
+      {
+        path: TRIGGER_USER_INTERACTION_METADATA_API_ROUTE,
```

Comment:
This path is `/api/intercept/user_interaction/{triggerId}`, which is a public API. I'm not sure the API needs to be public, because I don't see the need for customers to access this API programmatically. I think it will only be called from within the Kibana UI, right? If it is internal rather than public, the path should be `/internal/api/intercept/user_interaction/{triggerId}`. That would have a benefit as we would be free to make breaking changes to the API if we ever needed to.

Reply:
++ routes should be internal if not intended for external consumption.

Reply:
Yes... it would be called from the Kibana UI, thinking about this again indeed it should be internal, I'll make adjustments accordingly. Thanks for the pointer.